homeassistant/components/sensor/smappee.py

```diff
@@ -0,0 +1,177 @@
+    'alwaysOn': ['Always On', 'mdi:gauge', 'remote', 'W'],
+    'active_power': ['Active Power', 'mdi:power-plug', 'local', 'W'],
+    'current': ['Current', 'mdi:gauge', 'local', 'Amps'],
+    'voltage': ['Voltage', 'mdi:gauge', 'local', 'V'],
+    'active_cosfi': ['Power Factor', 'mdi:gauge', 'local', '%'],
+    'solar_today': ['Solar Today', 'mdi:white-balance-sunny', 'remote', 'kW'],
+    'power_today': ['Power Today', 'mdi:power-plug', 'remote', 'kW']
+}
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Smappee sensor."""
+    smappee = hass.data[DATA_SMAPPEE]
+
+    dev = []
+    if smappee.is_remote_active:
+        for sensor in SENSOR_TYPES:
+            if 'remote' in SENSOR_TYPES[sensor]:
+                for location_id in smappee.locations.keys():
+                    dev.append(SmappeeSensor(smappee, location_id, sensor))
+
+    if smappee.is_local_active:
+        for sensor in SENSOR_TYPES:
+            if 'local' in SENSOR_TYPES[sensor]:
+                if smappee.is_remote_active:
+                    for location_id in smappee.locations.keys():
+                        dev.append(SmappeeSensor(smappee, location_id, sensor))
+                else:
+                    dev.append(SmappeeSensor(smappee, None, sensor))
+    add_devices(dev)
+
+
+class SmappeeSensor(Entity):
+    """Implementation of a Smappee sensor."""
+
+    def __init__(self, smappee, location_id, sensor):
+        """Initialize the sensor."""
+        self._smappee = smappee
+        self._location_id = location_id
+        self._sensor = sensor
+        self.data = None
+        self._state = None
+        self._timestamp = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self._location_id:
+            location_name = self._smappee.locations[self._location_id]
+        else:
+            location_name = 'Local'
+
+        return "{} {} {}".format(SENSOR_PREFIX,
+                                 location_name,
+                                 SENSOR_TYPES[self._sensor][0])
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return SENSOR_TYPES[self._sensor][1]
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return SENSOR_TYPES[self._sensor][3]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attr = {}
+        if self._location_id:
+            attr['Location Id'] = self._location_id
+            attr['Location Name'] = self._smappee.locations[self._location_id]
+        attr['Last Update'] = self._timestamp
+        return attr
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
Instead of this, just define SCAN_INTERVAL and this method will never be called more than once during that interval.